platform-sdk/swirlds-component-framework/src/main/java/com/swirlds/component/framework/schedulers/internal/ConcurrentTask.java

```diff
@@ -61,14 +61,22 @@ class ConcurrentTask extends AbstractTask {
      * {@inheritDoc}
      */
     @Override
-    protected boolean exec() {
+    protected boolean onExecute() {
         try {
             handler.accept(data);
         } catch (final Throwable t) {
-            uncaughtExceptionHandler.uncaughtException(Thread.currentThread(), t);
+            completeExceptionally(t);
```

Comment:
Is it a duplicate now?

Reply:
I don't think so. If an exception is thrown here, it will be sent to uncaughtExceptionHandler from `onException()`, which is called from `completeExceptionally()`, but the task will still complete successfully (`return true` below). This is exactly how ConcurrentTask works today. I don't know if it's right or wrong.

If `ConcurrentTask.onExecute()` is changed to not catch any exceptions and let `exec()` catch everything, code flow will be slightly different. First, the task will fail (`exec()` will return false). Second, exception handler will be called after `offramp.offRamp()`, while today it is called before.

Reply:
Discussed it with Oleg offline, we both think this change (catch after finally vs finally after catch) is acceptable